pkg/object/s3.go

```diff
@@ -353,6 +377,8 @@ func newS3(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
 	}
 	if ep != "" {
 		awsConfig.Endpoint = aws.String(ep)
+	}
+	if uri.Path != "" {
```

Comment:
For backwards compatibility,  we want to use S3ForcePathStyle with a virtual hosted-style bucket.